plain/src/main/java/com/esewa/internship/gyalbusherpa/ProductUnitTestingInJUnit4/src/test/java/org/service/ProductServiceTest.java

```diff
@@ -0,0 +1,119 @@
+package org.service;
+
+import org.dao.ProductDao;
+import org.entity.Product;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertNotNull;
+
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.times;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ProductServiceTest {
+
+    @Mock
+    private ProductDao productDao;
+
+    @InjectMocks
+    private ProductService productService;
+
+    @Test
+    public void testGetProductById() {
+        Product product = new Product(1, "Product 1", 10.0);
+        when(productDao.getProductById(1)).thenReturn(product);
+
+        assertNotNull(productService.getProductById(1));
+        assertEquals(product, productService.getProductById(1));
```

Comment:
This line tests everything you tested before and after this.

Reply:
extra assert test removed